File: Civitas - Java/src/civitas/CasillaJuez.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author juanc
 */
public class CasillaJuez extends Casilla {
    
    //Tipo JUEZ
    CasillaJuez(int numCasillaCarcel, String nombre){
        super(nombre);
        Casilla.carcel = numCasillaCarcel;
    }
    
    //Se encarcela al jugador actual
    @Override
    protected void recibeJugador(int iactual, ArrayList<Jugador> todos){
        if( jugadorCorrecto(iactual, todos) == true ){
            informe(iactual, todos);
            todos.get(iactual).encarcelar(carcel);
        }
    }
    

}

File: Civitas - Ruby/lib/casilla_juez.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class Casilla_juez < Casilla
    def initialize(casilla_carcel, nombre)
      super(nombre)
      @carcel = casilla_carcel
    end
    
    #Se encarcela al jugador actual
    def recibe_jugador(iactual, todos)
      if jugador_correcto(iactual, todos)
        informe(iactual, todos)
        todos[iactual].encarcelar(@carcel)
      end
    end
    
  end
end
